core/src/apps/base.py

```diff
@@ -238,7 +239,16 @@ async def handle_GetFeatures(msg: GetFeatures) -> Features:
 
 
 async def handle_Cancel(msg: Cancel) -> Success:
-    workflow.close_others()
+    close_others = True
+    if storage_cache.homescreen_shown is storage_cache.HOMESCREEN_ON:
+        # Homescreen is shown and only this task is running
+        if len(workflow.tasks) == 1 and next(iter(workflow.tasks)).is_running():
+            # Prevent homescreen flickering on Cancel
+            close_others = False
+
+    if close_others:
+        workflow.close_others()
```

Comment:
how about instead you add an argument `workflow.close_others(close_default: bool = True)` and use `close_default=False` here

that way this weirdly specific logic is correctly hidden in the `workflow` module

Reply:
Thanks! Fixed in 4ee8985408.